sdk/python/lib/pulumi/resource.py

```diff
@@ -586,6 +589,26 @@ def _merge_lists(dest, source):
     return dest + source
 
 
+def _append_input_lists(a: Optional['Input[List[T]]'],
```

Comment:
What I find the SDK lacking is something along these lines:

```
map_input : Callable[[A],B] -> Input[A] -> Input[B]
map2_inputs : Callable[[A,B],C] -> Input[A] -> Input[B] -> Input[C]
```

This functions would try to preserve promptness, dropping to Output only if really necessary (which is what we are trying to do here). Could use it to zip the inputty lists then:

```
map2_inputs((lambda xs, ys: xs + ys), a , b)
```